pkg/cli/pod/pod.go

```diff
@@ -0,0 +1,491 @@
+		// we need to consider both the vcjob pod and other workload pods
+
+		// first, list all vcjobs's pods belong to the queue
+		labelSelector, err := createPodLabelSelectorByQueueName(v1alpha1.QueueNameKey, listPodFlags.QueueName)
+		if err != nil {
+			return err
+		}
+		listVcjobPodsRes, err := listPodFunc(labelSelector)
+		if err != nil {
+			return err
+		}
+		pods.Items = append(pods.Items, listVcjobPodsRes.Items...)
+
+		// then, list all other workload pods belong to the queue
+		listNormalPodsRes, err := listPodFunc(labels.Everything())
+		if err != nil {
+			return err
+		}
+		// filter normal pods by queue name
+		listNormalPodsRes =
+			filterFunc(listNormalPodsRes, schedulingv1beta1.QueueNameAnnotationKey, listPodFlags.QueueName)
+		// append only not exist
+		pods.Items = appendIfNotExists(pods.Items, listNormalPodsRes.Items)
+	} else {
+		// if neither job name nor queue name are specified,
+		// use default label selector, for all vcjobs's pods
+		labelSelector, err := createDefaultLabelSelector()
+		if err != nil {
+			return err
+		}
+		listPodsRes, err := listPodFunc(labelSelector)
+		if err != nil {
+			return err
+		}
+		pods.Items = append(pods.Items, listPodsRes.Items...)
+	}
+
+	if err != nil {
+		return err
+	}
+
+	if len(pods.Items) == 0 {
+		fmt.Printf("No resources found\n")
+		return nil
+	}
+	PrintPods(&pods, os.Stdout)
+
+	return nil
+}
+
+func PrintPods(pods *corev1.PodList, writer io.Writer) {
+	maxNameLen := 0
+	maxReadyLen := 0
+	maxStatusLen := 0
+	maxRestartLen := 0
+	maxAgeLen := 0
+
+	var infoList []PodInfo
+	for _, pod := range pods.Items {
+		info := printPod(&pod)
+		infoList = append(infoList, info)
+		// update max length for each column
+		if len(info.Name) > maxNameLen {
+			maxNameLen = len(info.Name)
+		}
+		if len(info.ReadyContainers) > maxReadyLen {
+			maxReadyLen = len(info.ReadyContainers)
+		}
+		if len(info.Status) > maxStatusLen {
+			maxStatusLen = len(info.Status)
+		}
+		if len(info.Restarts) > maxRestartLen {
+			maxRestartLen = len(info.Restarts)
+		}
+		if len(info.CreationTimestamp) > maxAgeLen {
+			maxAgeLen = len(info.CreationTimestamp)
+		}
+	}
+	columnSpacing := 8
+	maxNameLen += columnSpacing
+	maxReadyLen += columnSpacing
+	maxStatusLen += columnSpacing
+	maxRestartLen += columnSpacing
+	maxAgeLen += columnSpacing
+	formatStr := fmt.Sprintf("%%-%ds%%-%ds%%-%ds%%-%ds%%-%ds\n", maxNameLen, maxReadyLen, maxStatusLen, maxRestartLen, maxAgeLen)
+	_, err := fmt.Fprintf(writer, formatStr, Name, Ready, Status, Restart, Age)
+	if err != nil {
+		fmt.Printf("Failed to print Pod information: %s.\n", err)
+		return
+	}
+	for _, info := range infoList {
+		_, err := fmt.Fprintf(writer, formatStr, info.Name, info.ReadyContainers, info.Status, info.Restarts, info.CreationTimestamp)
+		if err != nil {
+			fmt.Printf("Failed to print Pod information: %s.\n", err)
+			return
+		}
+	}
+}
+
+// createPodLabelSelectorByJobName creates a label selector for selecting child pods of a vcjob.
```

Comment:
// createPodLabelSelectorByJobName creates a label selector for selecting child pods of a vcjob. => // createPodLabelSelectorByJobName creates a label selector for selecting pods belong to a vcjob.